src/dialects/mssql/query-generator.js

```diff
@@ -873,6 +873,114 @@ export class MsSqlQueryGenerator extends MsSqlQueryGeneratorTypeScript {
   selectFromTableFragment(options, model, attributes, tables, mainTableAs, where) {
     this._throwOnEmptyAttributes(attributes, { modelName: model && model.name, as: mainTableAs });
 
+    const dbVersion = this.sequelize.options.databaseVersion;
+    const isSQLServer2008 = semver.valid(dbVersion) && semver.lt(dbVersion, '11.0.0');
+
+    // mssql overwrite the abstract selectFromTableFragment function.
+    if (options.maxExecutionTimeHintMs != null) {
+      throw new Error(`The maxExecutionTimeMs option is not supported by ${this.dialect.name}`);
+    }
+
+    if (isSQLServer2008 && options.offset) {
```

Comment:
This whole if-statement, including the above constants of `dbVersion` and `isSQLServer2008` has been removed from the codebase. Can you also remove it from your changes?